src/Functions/density.cpp

```diff
@@ -0,0 +1,235 @@
+/*
+ *  This file is copied from R core to avoid including whole R core.
+ *
+ *  Copyright (C) 1997--2019   The R Core Team
+ *  Copyright (C) 1995, 1996   Robert Gentleman and Ross Ihaka
+ *
+ *  This program is free software; you can redistribute it and/or modify
+ *  it under the terms of the GNU General Public License as published by
```

Comment:
We cannot accept the code under GPL licenses. MIT, BSD, Apache should be OK, if I recall correctly. You'll have to find another implementation, or write this code yourself.